core/src/main/java/org/bitcoinj/wallet/BasicKeyChain.java

```diff
@@ -475,6 +479,9 @@ public BasicKeyChain toEncrypted(KeyCrypter keyCrypter, KeyParameter aesKey) {
                     throw new KeyCrypterException("The key " + key.toString() + " cannot be successfully decrypted after encryption so aborting wallet encryption.");
                 encrypted.importKeyLocked(encryptedKey);
             }
+            for (ListenerRegistration<KeyChainEventListener> listener : listeners) {
+                encrypted.listeners.add(listener);
```

Comment:
You could use the newly added `addEventListener(ListenerRegistration)` here, no?

Reply:
done.